docs/user-guide/a2-03-wgsl-target-specific.md

```diff
@@ -0,0 +1,158 @@
+---
+layout: user-guide
+permalink: /user-guide/wgsl-target-specific
+---
+
+WGSL specific functionalities
+=============================
+
+This chapter provides information for WGSL (WebGPU Shading Language) -specific functionalities and behaviors.
+
+
+System-Value semantics
+----------------------
+
+The system-value semantics are translated to the following WGSL code.
+
+| SV semantic name | WGSL code |
+|--|--|
+| SV_Barycentrics | *Not supported* |
+| SV_ClipDistance<N> | `@builtin(clip_distances)` |
```

Comment:
We need to make sure these are actually implemented. Right now we only support 5 semantics:
SV_Position, SV_DispatchThreadID, SV_GroupID, SV_GroupIndex, SV_GroupThreadID.

Reply:
Just create a PR and add these cases to `slang-ir-wgsl-legalize.cpp::getSystemValueInfo()`.

Reply:
Here is the PR:
https://github.com/shader-slang/slang/pull/5589

For the moment I marked clip planes as not supported, even though they are supported by a feature you can enable in WebGPU.

Currently looking into implementing that -- it seems like it should be pretty easy. I just need to understand how to get the semantic index so that I can get N for the array size of the parameter that the type takes...